app/backend/app.py

```diff
@@ -103,8 +111,15 @@ async def chat():
         impl = current_app.config[CONFIG_CHAT_APPROACHES].get(approach)
         if not impl:
             return jsonify({"error": "unknown approach"}), 400
-        r = await impl.run(request_json["history"], request_json.get("overrides") or {})
-        return jsonify(r)
+        should_stream = request_json.get("should_stream", False)
+        response_generator = impl.run(request_json["history"], request_json.get("overrides", {}), should_stream)
+        if should_stream:
+            response = await make_response(format_as_ndjson(response_generator))
+            response.timeout = None
+            return response
+        else:
+            # anext() is nicer but only available in Python 3.10
+            return jsonify(await response_generator.__anext__())
```

Comment:
this will only call `__anext__` once though, is that what we need? Doesn't it need to exhaust the iterator?